tests/appsec/test_automated_login_events.py

```diff
@@ -508,6 +528,52 @@ def test_login_sdk_failure_local(self):
             assert meta["appsec.events.users.login.failure.usr.exists"] == "true"
             assert_priority(span, meta)
 
+    def setup_login_success_headers(self):
+        self.r_hdr_success = weblog.post(
+            "/login?auth=local",
+            data={self.username_key: self.USER, self.password_key: self.PASSWORD},
+            headers=self.HEADERS,
+        )
+
+    @missing_feature(library="dotnet")
+    @missing_feature(library="java")
+    @missing_feature(library="nodejs")
+    @missing_feature(library="python")
+    @missing_feature(library="php")
+    @missing_feature(library="ruby")
+    def test_login_success_headers(self):
+        # Validate that all relevant headers are included on user login success on extended mode
+
+        def validate_login_success_headers(span):
+            for header, _ in self.HEADERS.items():
```

Comment:
I guess you are only validating the root span ? 

```suggestion
            if span.get("parent_id") not in (0, None):
                return
                
            for header, _ in self.HEADERS.items():
```

Yes, we miss some helper here...